### PR TITLE
Fix: size_fd has to be read under a lock

### PR DESCRIPTION
**Pull request by ldorau**
### Description

`size_fd` has to be read under a lock.

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Review by lukaszstolarczuk — COMMENTED** [review]
👍

**Review by pbalcer — APPROVED** [review]
That's a misleading commit message :) `fd_size` needs to be read under a lock, not `fd_offset` (a stack variable) set under one.

File: src/provider/provider_os_memory.c [hunk 1]
```diff
@@ -637,7 +637,8 @@ static inline void assert_is_page_aligned(uintptr_t ptr, size_t page_size) {
 static int os_mmap_aligned(void *hint_addr, size_t length, size_t alignment,
```
Comment (lplewa) [inline src/provider/provider_os_memory.c:637]: NIT: 12th argument to this function. This screams for refactor.

Comment (ldorau) [reply src/provider/provider_os_memory.c:637]: Yes, I will do that in a separate PR.

**Comment by ldorau** [comment]
> That's a misleading commit message :) `fd_size` needs to be read under a lock, not `fd_offset` (a stack variable) set under one.

Fixed.

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**